diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -306,20 +310,18 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
                     getAllLeafsWithSpecificType(SAFE_ACCESS, it)
                 }
             }
-            // fixme: we can't distinguish fully qualified names from chain of calls for now
             ?.filter {
-                if (it.elementType == DOT) {
-                    val dotExprs = it.treeParent.text.split(".")
-
-                    // should skip node's that are val params. For example: b.some(z.a()) -> skip z.a()
-                    if (it.treeParent.treeParent != null && it.treeParent.treeParent.elementType == VALUE_PARAMETER_LIST)
-                        return@filter false
-
-                    val firstCallee = 1
-                    return@filter if (dotExprs[firstCallee].contains('(') || dotExprs[firstCallee].contains('{')) {
-                        true
-                    } else dotExprs.last().contains('(') || dotExprs.last().contains('{')
-                }
+                // should skip node's that are val params. For example: b.some(z.a()) -> skip z.a()
+                if (it.treeParent.treeParent != null && it.treeParent.treeParent.elementType == VALUE_ARGUMENT)
+                    return@filter false
+                true
+            }
+            ?.dropWhile { !it.treeParent.textContains('(') && !it.treeParent.textContains('{') }
+            // fixme: we can't distinguish fully qualified names from chain of calls for now
+            ?.filterIndexed { index, it ->
+                // if first callee is multiline lambda, then we let user decide how to line lambda
+                if (index == 0 && isMultilineLambda(it.treeParent))
+                    return@filterIndexed false
```

Comment:
Why do you use `return` here? You can just write a predicate, it would be more readable.